app/helpers/application_helper.rb

```diff
@@ -7,4 +7,14 @@ def signup_path?
 
     false
   end
+
+  def user_like_post?(one, two, three)
+    return true if Like.where(status: one).where(user_id: two).where(post_id: three).exists?
```

Comment:
- [ ] There should be no application logic in helpers or in views. 
The line `Like.where(status: one).where(user_id: two).where(post_id: three).exists?` belongs to the user model because it directly hit the database level, helpers and views only display data which are available to them from the controllers.
[OPTIONAL] We can take advantage of has_many: through association to define what posts that a specific user liked with: `has_many :liked_posts_, through: :likes, source: 'Post'` in user model, this way we can simply check if the user liked a post with `user.liked_post.ids.include?(post.id)`. You can even wrap it in user model instance method.